esmvalcore/cmor/_fixes/icon/_base_fixes.py

```diff
@@ -320,9 +363,7 @@ def _get_grid_from_cube_attr(self, cube: iris.cube.Cube) -> iris.cube.Cube:
                 grid_path,
             )
 
-        self._horizontal_grids[grid_name] = self._load_cubes(grid_path)
-
-        return self._horizontal_grids[grid_name]
+        return self._load_cubes(grid_path)
```

Comment:
I'd move this one line up, and return a `cubes` etc, just in case it barfs, it's nasty to get None's or exceptions while returning

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2337/commits/9cd179832c5ef18763999dc17e6be8b2f8b6eb56